### ISSUE TITLE
0.13.2: New wildcard versions POM-format inside dependencies not resolved by sbt

### ISSUE THREAD

**Issue by helllamer** [sbt/sbt#1275]
I have problem with dependencies with sbt-0.13.2.
I have two projects: a shared published library `library1` and a project `project1`, that uses `library1` as dependency.
library1/build.sbt:

```
libraryDependencies ++= Seq(
  "org.gnu.inet" % "libidn" % "1.1+"
)
```

`sbt publish` on 0.13.1 generates pom file `library1_2.10-1.0.0-SNAPSHOT.pom` with wildcard versions like:

```
<dependency>
   <groupId>org.gnu.inet</groupId>
   <artifactId>libidn</artifactId>
   <version>1.1+</version>
</dependency>
```

This POM processed and resolved by clients with no problems.

0.13.2 generates different POM with different (invalid) version format:

```
<dependency>
    <groupId>org.gnu.inet</groupId>
    <artifactId>libidn</artifactId>
    <version>[1, .2)</version>
</dependency>
```

On dep.resolve inside `project1`, libidn version is not resolved by other sbt (0.13.1, 0.13.2 - does not matter):

```
$ sbt update
[info] Updating
{file:/Users/user1/project1/}project1...
[info] Resolving org.gnu.inet#libidn;[1, .2) ...
[warn]  module not found: org.gnu.inet#libidn;[1, .2)
[warn] ==== local: tried
[warn]  
/Users/user1/.ivy2/local/org.gnu.inet/libidn/[revision]/ivys/ivy.xml
[warn] ==== repo1: tried
[warn]  
http://repo1.maven.org/maven2/org/gnu/inet/libidn/[revision]/libidn-[revision].pom
[warn]   [1.15, 0.6.5]
[warn] ==== typesafe-releases: tried
[warn]  http://repo.typesafe.com/typesafe/releases/org/gnu/inet/libidn/[revision]/libidn-[revision].pom
...
```

Possibly, this problem related to #1136.

**Comment by jsuereth** [comment]
Just a quick note:

the `1.2+` syntax that ivy allows does not make a valid pom.xml file.   However, it looks like you've caught a bug in the expansion.

**Comment by helllamer** [comment]
> the 1.2+ syntax that ivy allows does not make a valid pom.xml file

I think a some warning/error or something else should be thrown to user in such cases. Now, 0.13.2 silently generates invalid pom.xml.

**Comment by jsuereth** [comment]
Yeah, it's supposed to error when it encounters a `+` it can't migrate.   However, it seems our test for this isn't quite robust enough:  https://github.com/sbt/sbt/blob/0.13/ivy/src/test/scala/MakePomTest.scala

We may just alter the `+` to issue errors in `makePom` about how it's invalid maven.   

Note: We're trying to clean up our bad behavior on maven-central causing issues for other libraries/tools.

**Comment by eed3si9n** [comment]
A note on `1.1+`.  This came up in a recent sbt-dev thread named [Dangerous new bug in sbt 0.13.2 - Maven POM dependency versioning](https://groups.google.com/d/msg/sbt-dev/7sE1885rk_g/dLhYm0YGI74J) too but I don't think people are using `+` correctly.

Ivy's [Fixed and dynamic revisions](http://ant.apache.org/ivy/history/latest-milestone/ivyfile/dependency.html) documentation says:

> - end the revision with a `+`<br>
>   selects the latest sub-revision of the dependency module. For instance, 
>   if the dependency module exists in revision 1.0.3, 1.0.7 and 1.1.2, "1.0.+" will select 1.0.7.

The notation `1.1+` gives an impression that it's somehow "1.1 and above", but it's more like "the latest within 1.x series." So it should be `[1.0, 2.0)` or we throw our hands in the air and fail.

**Comment by eed3si9n** [comment]
sbt/sbt@5486daf7006462be13b82d6ae225e816f038c9ff fixed this issue in 0.13.5.
